content/concepts/connections.md

```diff
@@ -0,0 +1,155 @@
+---
+title: Connections
+weight: 3
+---
+
+A libp2p connection allows two peers to read and write data to each other. 
+Peers can connect through transport protocols. Transport protocols are core 
+abstractions of libp2p and offer extensibility. As a modular networking stack, 
+libp2p is transport-agnostic and does not enforce the implementation of a specific 
+transport protocol. Though the support of modern transport protocols is a primary 
+focus for libp2p, implementers may choose to support multiple types of transport. 
+Learn more about transport protocols in libp2p on the 
+[transport guide]((/concepts/transports/)). 
+
+Let's first outline key terminology to help us understand connections in libp2p.
+
+## Core libp2p components
+
+- **Peer**: a participant in a libp2p network.
+- **Peer-to-Peer (P2P)**: a distributed network in which workloads are 
+  shared between *Peers*.
+- **Connection**: a network layer connection between two peers in a libp2p 
+  network.
+- **Initiator**: a peer that initiates a connection through a dial request.
+- **Responder**: a peer that listens to dial requests.
+- **Transport**: a mechanism of communication that creates a *Connection*, 
+  e.g., TCP, QUIC, WebRTC, etc.
+- **Stream**: an abstraction on top of a *Connection* representing a bidirectional 
+  communication channel between two *Peers*.
+- **Multiplexer**: a mechanism that manages multiple streams and ensures parallel 
+  messages reach the correct *Stream*, based on stream identification.
+- **Secure channel**: establishes a secure, encrypted, and authenticated channel 
+  over a *Connection*.
+- **Upgrader**: a mechanism that takes a *Connection* returned by a *Transport*, 
+  and performs protocol negotiation to set up a secure, multiplexed channel on 
+  top of which *Streams* can be opened.
+
+## Opening a libp2p connection
+
+A libp2p connection only requires two core operations: dialing and listening. 
+Both operations exist as interfaces in libp2p, where a transport protocol is used 
+to expose the dialing and listening interfaces. The listening interface allows a 
+peer to listen to connection requests from others, while the dialing interface 
+allows a peer to send connection requests to a listening peer. Most of the connection 
+initation process is abstracted through libp2p's interfaces.
+
+### How do peers prevent multiple connections from being established? 
+
+An interface called a [switch](/concepts/stream-multiplexing/#switch/swarm) 
+(and sometimes swarm) implements a basic dial queue that manages an application's 
+dialing and listening state. This prevents multiple, 
+simultaneous dial requests to the same peer. 
```

Comment:
Sorry in case I sparked confusion here before. As a data point, rust-libp2p manages a dial queue in its `ConnectionPool` and not in its `Swarm`.

I would drop these implementation specific terms here entirely.

Just my opinion. Maybe others feel strongly about it.